data_model/src/events/data/filters.rs

```diff
@@ -0,0 +1,616 @@
+//! This module contains `EventFilter` and entities for filter
+
+use super::*;
+
+/// Create entity filter type. See usage below
+///
+/// This could be implemented with generic struct, but it's bad to have nested generics in filters
+/// public API (especially for schemas)
+macro_rules! entity_filter {
+    (pub struct $name:ident { event: $entity_type:ty, filter: $event_filter_type:ty, }) => {
+        entity_filter! {
+            $name,
+            $entity_type,
+            $event_filter_type,
+            concat!("Filter for ", stringify!($event_filter_type), " entity"),
+            concat!("Construct new ", stringify!($name))
+        }
+    };
+    ($name:ident, $entity_type:ty, $event_filter_type:ty, $struct_doc:expr, $new_doc:expr) => {
```

Comment:
if this is [internal **rule](http://danielkeep.github.io/tlborm/book/pat-internal-rules.html),** you could prepend `@`